### PR TITLE
[SYCL][LowerWGScope][SpecConstantsPass] fix new call's calling conv for spir/spirv

### PR DESCRIPTION
**Pull request by wenju-he**
Newly created non-kernel function and function call should have spirv_func calling convention.

### CODE REVIEW

**Comment by wenju-he** [comment]
one motivation is intel gpu backend requires calling conv to be consistent.

**Review by AlexeySachkov — APPROVED** [review]

File: llvm/lib/SYCLLowerIR/LowerWGScope.cpp [hunk 1]
```diff
@@ -1001,6 +1001,8 @@ Instruction *spirv::genWGBarrier(Instruction &Before, const Triple &TT) {
   FunctionCallee FC =
       M.getOrInsertFunction(Name, Attr, RetTy, ScopeTy, ScopeTy, SemanticsTy);
   assert(FC.getCallee() && "spirv intrinsic creation failed");
+  if (TT.isSPIR())
```
[the remark this answers is not in the document]
Comment (wenju-he) [reply llvm/lib/SYCLLowerIR/LowerWGScope.cpp]: done
Also fixed SpecConstantsPass

**Comment by asudarsa** [comment]
> one motivation is intel gpu backend requires calling conv to be consistent.

Thanks for adding this support. Looks good. One question. Is there documentation anywhere that talks about this calling convention and about which functions need this.

**Review by sarnex — APPROVED** [review]
lgtm as in my feedback is addressed, thx

**Comment by wenju-he** [comment]
> > one motivation is intel gpu backend requires calling conv to be consistent.
> 
> Thanks for adding this support. Looks good. One question. Is there documentation anywhere that talks about this calling convention and about which functions need this. Surely not a blocker. Thanks

@asudarsa For spir target, quote from page 33 of https://registry.khronos.org/SPIR/specs/spir_spec-2.0.pdf
```
3.8 Calling Conventions
SPIR kernels should use "spir kernel" calling convention. Non-kernel functions use "spir func"
calling convention. All other calling conventions are disallowed.
```
So I assume every function call should have either spir_kernel or spir_func calling convention.

I am not ware of any documentation of the requirement of consistent calling convention in the intel gpu backend.

**Comment by wenju-he** [comment]
@intel/llvm-gatekeepers please merge, thank you

**Pull request merged**